### ISSUE TITLE
Arrival at a creature is dispatched on how the aim was spelled, not on what the creature is: a flight-master errand cancels a training plan, and the auction leg can never hold at a counter

### ISSUE THREAD

**Issue by quadseven** [quadseven/mod-overseer#402]
## Arrival at a creature is dispatched on how the aim was SPELLED, not on what the creature is

Four handlers can run when a character arrives at a creature. Three of them ask the aim
string what to do and one asks the world. Only the one that asks the world works for an aim
that names a creature by entry, and the three that ask the string misfire in three different
ways.

The chain is `src/mod_overseer.cpp:15706-15742`:

```cpp
if (doorway)                                            { /* ... */ }
else if (escorted)                                      { /* ... */ }
else if (plan && !TrainOnArrival(name, bot, entry, *plan))   // :15717
{ /* hold position, not released */ }
else
{
    OverseerDecisions::CounterRole const counter =
        OverseerDecisions::CounterRoleForAim(target);   // :15760, reads the STRING
    ...
    DiscoverFlightPointOnArrival(name, bot, entry);     // :15857, reads the CREATURE
    LOG_INFO(... "errand done, releasing" ...);         // :15860
    _travelAims.Release(name);
}
```

## Correction: the discovery DID run, and the quoted log proves it

The reading that the entry errand is dispatched down the trainer path and rejected before
anything else can look at the creature is not what the code does, and the evidence for that
is in the same two log lines.

`TrainOnArrival` **returns `true`** on the "will not train" path, after clearing the learn
aim:

`src/mod_overseer.cpp:10180-10187`

```cpp
Trainer::Trainer* trainer = sObjectMgr->GetTrainer(entry);
if (!trainer || !trainer->IsTrainerValidForPlayer(bot))
{
    LOG_WARN(... "will not train '{}' ... Dropping the errand" ...);
    ClearLearnAim(name);
    return true;                                   // :10186
}
```

`true` makes `!TrainOnArrival(...)` false, so the `else if` at `:15717` is **not** taken and
control falls into the `else`. The line `reached '{}' (creature {}) - errand done, releasing`
exists exactly once in this file, at `:15860`, and it sits **three statements after**
`DiscoverFlightPointOnArrival` at `:15857`. There is no path that prints it without having
called discovery first.

So discovery ran, reached the creature, and returned early. Every early return in it is
silent except one, and the one that fired is this:

`src/mod_overseer.cpp:10305-10306`

```cpp
if (bot->m_taxi.IsTaximaskNodeKnown(node))
    return;   // already known - the ordinary case after the first visit
```

**The character used in the test already holds node 80.** By the taximask reading recorded
on #388, two of the five hold it and three do not. The trial was run on one of the two. The
arithmetic checks out, incidentally: `PlayerTaxi.h:35-40` masks on `(nodeidx - 1)`, so node
80 is field `79 / 32` = 2, bit `79 % 32` = 15, value 32768, exactly as reported.

**So the mechanism already satisfies the acceptance for a character that lacks the node.**
What it does not do is say so, which is why a correct no-op was indistinguishable from a
broken one. That silence is the thing to fix, not the dispatch.

## The real defect the trial exposed

Walking a character to a flight master **destroyed its training plan.**

`TrainOnArrival` is called on every creature arrival whenever the character has a learn plan
at all, with no regard for what the travel errand was for. Bragok is not a trainer, so it
logged a warning about an errand nobody issued and called `ClearLearnAim` at `:10185`.

The learn aim is a separate column from the travel aim. Sending a character anywhere near
any non-trainer creature while it has a profession queued silently cancels that profession.
Every flight-master errand this module is about to start issuing would do this, once per
trip.

## Which brings the auction and repair legs in, because it is one root cause

`CounterRoleForAim` matches exactly three whole strings:

`src/overseer_decisions.cpp`

```cpp
if (aim == "vendor")   return CounterRole::Vendor;
if (aim == "banker")   return CounterRole::Banker;
if (aim == "repair")   return CounterRole::Repairer;
return CounterRole::None;
```

Two consequences, both live:

**1. `auctioneer` is not in that list at all.** It is one of the thirteen keywords
`TravelRoles()` resolves (`src/mod_overseer.cpp:9181`), so a character can be *sent* to an
auctioneer and does arrive. But the role comes back `None`, `CounterArrivalStep` answers
`Done` on its first line, no `HoldAtTheCounter` is taken, and the errand releases. The
release is the signal the outside pass reads as "arrived" before it writes a row, and the
character has an AI tick to wander off before the rows land. That is a complete explanation
for the auction leg never having produced one, and it needs no second cause.

**2. An entry-id aim yields `None` for every role.** `"16227"` is not `"vendor"`. So aiming
at a repair vendor by entry gets no counter hold, while aiming at the keyword `repair` does.
The two forms of the same errand behave differently, and only one of them is tested.

The repair failure recorded on #398 has its own separate cause and is not this: that errand
tests whether a repairer is already inside the interaction gate and gives up, rather than
travelling to one. This is a second, independent way the same leg can fail once it does
travel.

## Why discovery is the one handler that works

Because it asks the world instead of the string:

`src/mod_overseer.cpp:10290-10291`

```cpp
if (!npc->HasNpcFlag(UNIT_NPC_FLAG_FLIGHTMASTER))
    return;
```

That is the pattern the other three should follow. A creature's own npcflags are what decide
what standing next to it can mean; the aim string only says why the character was sent.

## What to change

All of the deciding is pure and belongs in `src/overseer_decisions.{h,cpp}`.

- `CounterRoleForAim` learns `auctioneer`, and `CounterRole` gains `Auctioneer`.
- A new `CounterRoleForNpcFlags(npcFlags)` derives the role from the creature itself, used
  when the aim yields `None`. Precedence is documented rather than incidental: repairer,
  banker, auctioneer, then vendor, because the first three are narrow capabilities and
  vendor is the broad one they usually co-occur with, so it is the fallback rather than the
  winner. A repairer that also sells is held as a repairer.
- A new `ArrivalAnswersLearnAim(aimNamesTrainerRole, creatureTrains)` decides whether this
  arrival is entitled to consume the learn aim. True when the errand was a trainer errand,
  true when the creature actually trains, false otherwise. A flight master does not get to
  cancel a profession.
- `DiscoverFlightPointOnArrival` says what it did on every path, including the no-ops.

## Acceptance criteria

- [x] `CounterRoleForAim("auctioneer")` returns `Auctioneer`, and the three existing
      keywords are unchanged.
- [x] `CounterRoleForNpcFlags` returns the narrow role for a creature carrying several, in
      the documented precedence, and `None` for a creature carrying no counter flag at all.
- [x] A flight master's flags yield `None`, so discovery-only arrivals are not given a
      counter hold they have no rows for.
- [x] `ArrivalAnswersLearnAim` is false for a non-trainer creature reached on a non-trainer
      aim, so a learn aim survives a flight-master errand.
- [x] `ArrivalAnswersLearnAim` is true when the aim named a trainer role, and true when the
      creature trains, so today's behaviour is kept for every errand that is about training.
- [x] The adapter consults the creature's flags only when the aim yields `None`, so a
      keyword aim keeps meaning exactly what it means today.
- [x] `DiscoverFlightPointOnArrival` speaks on every outcome **after** the flight-master
      gate: no node of our team here, already known, learned, and did not stick. The
      "already known" case is distinguishable from "no node here".

  *(Corrected from "all five outcomes" as first written, which was wrong and would have made
  the log unusable. This function runs opportunistically on EVERY creature arrival, so the
  two checks before the gate - "nothing of that entry is standing here" and "this is not a
  flight master" - fire for every vendor, trainer and banker the family ever walks to. Those
  two must stay silent; they are not outcomes of a discovery attempt, they are the function
  declining to be one.)*
- [x] Every pre-existing test still passes, `test_counter_hold.cpp` included.

## Out of scope

- The repair errand's own failure to travel, #398. Separate cause, separate fix.
- The name lookup, #398. Aiming by entry id works and is what the flight errand should use.
- Which character goes to which flight master and when, #388. This makes the errand safe to
  issue; it does not schedule it.
- The guarded-ground pricing in #401, which is independent and already open.

**Comment by quadseven** [comment]
Ticked against #403, with one criterion **corrected rather than ticked as written**.

The seventh box originally asked that `DiscoverFlightPointOnArrival` log on "all five outcomes". That was wrong and would have made the log unusable: this function runs opportunistically on every creature arrival, so the two checks before the flight-master gate ("nothing of that entry is standing here" and "this is not a flight master") fire for every vendor, trainer and banker the family ever walks to. Those two stay silent. They are not outcomes of a discovery attempt; they are the function declining to be one. The criterion now asks for the four outcomes after the gate, which is what shipped.

What proves each of the others, in `tests/test_arrival_meaning.cpp` unless noted:

| criterion | test |
|---|---|
| `auctioneer` resolves, three shipped keywords unchanged | `TheAuctioneerKeywordIsFinallyOneOfThem`, `TheThreeShippedKeywordsAreUnchanged`, and `OnlyTheFourEconomyKeywordsAreCounters` in `test_counter_hold.cpp` |
| precedence, and `None` for no counter flag | `NarrowBeatsBroadWhenACreatureIsSeveralThings`, asserted on the real 4224 spawn, and `ACreatureThatIsNoCounterGetsNoHold` |
| a flight master yields `None` | `ACreatureThatIsNoCounterGetsNoHold`, on the real 8195 |
| a learn aim survives a flight-master errand | `AFlightMasterDoesNotCancelAProfession` |
| today's behaviour kept for errands that are about training | `AnArrivalThatIsAboutTrainingStillAnswersIt`, both halves |
| the adapter consults flags only when the aim answers `None` | structural, not assertable through the pure API: `aimRole != None ? aimRole : CounterRoleForNpcFlags(...)` is the one call site and it is readable in the diff |
| every pre-existing test still passes | all 71 `tests/test_*.cpp` compile and run green, `-Wall -Wextra` clean |

Writing the flag-value test caught a real error before it shipped: `NPC_FLAG_BANKER` is `0x00020000`, not the `0x00040000` I first wrote from memory. That is why the values are asserted against arithmetic on two spawns this module has actually stood next to rather than merely commented.
